src/meta/MetaHttpHandler.cpp

```diff
@@ -49,22 +79,49 @@ void MetaHttpHandler::onEOM() noexcept {
                 .status(405, "Method Not Allowed")
                 .sendWithEOM();
             return;
+        case HttpCode::E_ILLEGAL_ARGUMENT:
+            ResponseBuilder(downstream_)
+                .status(404, "Illegal Argument")
+                .sendWithEOM();
+            return;
         default:
             break;
     }
 
-    // read meta daemon status
-    folly::dynamic vals = getStatus();
-    if (returnJson_) {
-        ResponseBuilder(downstream_)
-            .status(200, "OK")
-            .body(folly::toJson(vals))
-            .sendWithEOM();
+    if (method == "status") {
+        folly::dynamic vals = getStatus();
+        if (returnJson_) {
+            ResponseBuilder(downstream_)
+                .status(200, "OK")
+                .body(folly::toJson(vals))
+                .sendWithEOM();
+        } else {
+            ResponseBuilder(downstream_)
+                .status(200, "OK")
+                .body(toStr(vals))
+                .sendWithEOM();
+        }
+    } else if (method == "download") {
+        if (auto hadoopHome = std::getenv("HADOOP_HOME")) {
```

Comment:
It is going to be hard to modify at runtime if you use ENV as config.

Reply:
`HADOOP_HOME` is the standard settings for `Hadoop`. 



> any tests?

I have deploy a mini hadoop cluster on dev environment. It's work fine. :)